src/Parsers/isUnquotedIdentifier.h

```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+#include <base/types.h>
+
+namespace DB
+{
+
+bool isUnquotedIdentifier(const String & name);
```

Comment:
Maybe just `isIdentifier` is enough? And a comment seems to be useful here (I know this function is pretty simple, but nevertheless).

Reply:
"Unquoted" is important here. `1` is also an identifier but should not be used as name.

Reply:
`1` is not an identifier until it's quoted with backticks. Probably a longer name could be used here, like `isValidIdentifier()`. But I won't insist.